src/url.rs

```diff
@@ -150,6 +154,46 @@ impl PyUrl {
     fn __getnewargs__(&self) -> (&str,) {
         (self.__str__(),)
     }
+
+    #[classmethod]
+    #[pyo3(signature=(*, scheme, host, user=None, password=None, port=None, path=None, query=None, fragment=None))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn build<'a>(
+        cls: &'a PyType,
+        scheme: &str,
+        host: &str,
+        user: Option<&str>,
+        password: Option<&str>,
+        port: Option<u16>,
+        path: Option<&str>,
+        query: Option<&str>,
+        fragment: Option<&str>,
+    ) -> PyResult<&'a PyAny> {
+        let user_password = match (user, password) {
```

Comment:
@davidhewitt I wasn't sure if we should use MultiHostUrl here, felt a bit... wrong? maybe the naming isn't right for that Struct if it should be used here.

Also had an issue passing `host` in since it isn't an Option<&str>, though i suppose thats an issue we can get around.

Reply:
Ah right, I see, I think it might now be:

```rust
            let url_host = MultiHostUrlHost {
                username: username.map(Into::into),
                password: password.map(Into::into),
                host: Some(host.into()),                
                port,
            };
```

We can call the struct `UrlHostParts` or something else if you prefer.

What I'm more concerned about is that we seem to have different logic for the `match` here versus the `impl Display for MultiHostUrlHost` below when deciding whether to include `@`.

Reply:
Yes - thanks for spotting that, i'll address